nexus/db-queries/src/db/pool.rs

```diff
@@ -3,108 +3,144 @@
-        )
+impl SingleHostResolver {
+    fn new(config: &DbConfig) -> Self {
+        let backends = Arc::new(BTreeMap::from([(
+            backend::Name::new("singleton"),
+            backend::Backend { address: config.url.address() },
+        )]));
+        let (tx, _rx) = watch::channel(backends.clone());
+        Self { tx }
     }
+}
 
-    fn new_builder(
-        log: &slog::Logger,
-        db_config: &DbConfig,
-        builder: bb8::Builder<ConnectionManager<DbConnection>>,
-    ) -> Self {
-        let url = db_config.url.url();
-        let log = log.new(o!(
-            "database_url" => url.clone(),
-            "component" => "db::Pool"
-        ));
-        info!(&log, "database connection pool");
-        let error_sink = LoggingErrorSink::new(log);
-        let manager = ConnectionManager::<DbConnection>::new(url);
-        let pool = builder
-            .connection_customizer(Box::new(
-                super::pool_connection::ConnectionCustomizer::new(),
-            ))
-            .error_sink(Box::new(error_sink))
-            .build_unchecked(manager);
-        Pool { pool }
+impl Resolver for SingleHostResolver {
+    fn monitor(&mut self) -> watch::Receiver<AllBackends> {
+        self.tx.subscribe()
     }
+}
 
-    /// Returns a reference to the underlying pool.
-    pub fn pool(&self) -> &bb8::Pool<ConnectionManager<DbConnection>> {
-        &self.pool
-    }
+fn make_dns_resolver(
+    bootstrap_dns: Vec<SocketAddr>,
+) -> qorb::resolver::BoxedResolver {
+    Box::new(DnsResolver::new(
+        service::Name(internal_dns::ServiceName::Cockroach.srv_name()),
+        bootstrap_dns,
+        DnsResolverConfig {
+            query_interval: tokio::time::Duration::from_secs(10),
+            hardcoded_ttl: Some(tokio::time::Duration::MAX),
+            ..Default::default()
+        },
+    ))
 }
 
-#[derive(Clone, Debug)]
-struct LoggingErrorSink {
-    log: slog::Logger,
+fn make_single_host_resolver(
+    config: &DbConfig,
+) -> qorb::resolver::BoxedResolver {
+    Box::new(SingleHostResolver::new(config))
 }
 
-impl LoggingErrorSink {
-    fn new(log: slog::Logger) -> LoggingErrorSink {
-        LoggingErrorSink { log }
-    }
+fn make_postgres_connector() -> qorb::backend::SharedConnector<QorbConnection> {
+    // Create postgres connections.
+    //
+    // We're currently relying on the DieselPgConnector doing the following:
+    // - Disallowing full table scans in its implementation of "on_acquire"
+    // - Creating async_bb8_diesel connections that also wrap DTraceConnections.
+    let user = "root";
+    let db = "omicron";
+    let args = Some("sslmode=disable");
+    Arc::new(DieselPgConnector::new(user, db, args))
 }
 
-impl bb8::ErrorSink<ConnectionError> for LoggingErrorSink {
-    fn sink(&self, error: ConnectionError) {
-        error!(
-            &self.log,
-            "database connection error";
-            "error_message" => #%error
-        );
+impl Pool {
```

Comment:
nit, not important: personally, i'm not super convinced that these all need `_qorb_` in their names; that seems kinda like an implementation detail. as a user of this module, I feel like i mostly just care about the behavior of the pool, not which library is used to provide it.

Reply:
Fair enough, I think it was way more relevant during implementation when I had a combination of bb8 and qorb (briefly) but that is no longer the case. I'll update these names.

Reply:
:woman_shrugging: it's up to you, i don't feel terribly strongly about this!

Reply:
I think it was the right call! Updated 'em